### PR TITLE
Account has-virtual-mfa filter

### PR DESCRIPTION
**Pull request by Sutto**
Again, part of the CIS Foundation Benchmark for AWS - Checks for accounts where the root account has a virtual MFA. this is first attempt at implementing it.

It's super simple at the moment, but may be useful to make it a ValueFilter instead that operates on the object - or a related filter (especially if we complete #1144) - but there is a little bit of things around the edges to consider here.

### CODE REVIEW

**Comment by kapilt** [comment]
there's already a multiple mechanisms to determine this via custodian (credential report on account or iamsummary).

**Comment by Sutto** [comment]
They test for MFA's in general, don't they? This specifically is virtual MFA devices - the use case being requiring the root account (Ergo one with most access) to use a hardware MFA device.

File: c7n/resources/account.py [hunk 1]
```diff
@@ -656,3 +656,42 @@ def process(self, accounts):
...
+
+    def mfa_belongs_to_root_account(self, mfa):
+        return mfa['SerialNumber'].endswith(':mfa/root-account-mfa-device')
+
+    def account_has_virtual_mfa(self, account):
+        if not account.get('c7n:VirtualMFADevices'):
+            client = local_session(self.manager.session_factory).client('iam')
+            paginator = client.get_paginator('list_virtual_mfa_devices')
+            raw_list = paginator.paginate().build_full_result()['VirtualMFADevices']
+            print(raw_list)
```
Comment (kapilt) [inline c7n/resources/account.py]: stray print

File: c7n/resources/account.py [hunk 2]
```diff
@@ -656,3 +656,42 @@ def process(self, accounts):
...
+    schema = type_schema('has-virtual-mfa', **{'value': {'type': 'boolean'}})
+
+    permissions = ('iam:ListVirtualMFADevices',)
+
+    def mfa_belongs_to_root_account(self, mfa):
+        return mfa['SerialNumber'].endswith(':mfa/root-account-mfa-device')
+
+    def account_has_virtual_mfa(self, account):
+        if not account.get('c7n:VirtualMFADevices'):
+            client = local_session(self.manager.session_factory).client('iam')
```
Comment (kapilt) [inline c7n/resources/account.py]: wondering if we should go ahead and turn this into an iam resource (`iam-virtual-mfa`)

**Comment by kapilt** [comment]
thanks that makes sense, and indeed the other techniques don't distinguish on mfa type.

**Pull request merged**